From c1b8934faaaa9a4be4831418abc0893e10f61faa Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 22 Aug 2012 13:16:23 +0200
Subject: [PATCH] znet: add some more kernel drivers

---
 modules.d/95znet/module-setup.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/95znet/module-setup.sh b/modules.d/95znet/module-setup.sh
index b2ffa10..406d139 100755
--- a/modules.d/95znet/module-setup.sh
+++ b/modules.d/95znet/module-setup.sh
@@ -14,7 +14,7 @@ depends() {
 }
 
 installkernel() {
-    instmods ctcm
+    instmods ctcm lcs qeth qeth_l2 qeth_l3
 }
 
 install() {
