From dff1671f8f21b88479ad022623ff26e40e958bd6 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 10 Sep 2012 15:49:37 +0200
Subject: [PATCH] iscsi/parse-iscsiroot.sh: delay loading of iscsi modules

---
 modules.d/95iscsi/parse-iscsiroot.sh | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/modules.d/95iscsi/parse-iscsiroot.sh b/modules.d/95iscsi/parse-iscsiroot.sh
index e7fc224..cc59f5a 100755
--- a/modules.d/95iscsi/parse-iscsiroot.sh
+++ b/modules.d/95iscsi/parse-iscsiroot.sh
@@ -53,12 +53,6 @@ if [ -n "$iscsiroot" ] ; then
     [ -z "$netroot" ] || [ "$netroot" = "iscsi" ] && netroot=iscsi:$iscsiroot
 fi
 
-modprobe -q qla4xxx
-modprobe -q cxgb3i
-modprobe -q cxgb4i
-modprobe -q bnx2i
-modprobe -q be2iscsi
-
 # iscsi_firmware does not need argument checking
 if [ -n "$iscsi_firmware" ] ; then
     netroot=${netroot:-iscsi}
@@ -69,6 +63,12 @@ fi
 # If it's not iscsi we don't continue
 [ "${netroot%%:*}" = "iscsi" ] || return
 
+modprobe -q qla4xxx
+modprobe -q cxgb3i
+modprobe -q cxgb4i
+modprobe -q bnx2i
+modprobe -q be2iscsi
+
 if [ -z "$iscsi_firmware" ] ; then
     type parse_iscsi_root >/dev/null 2>&1 || . /lib/net-lib.sh
     parse_iscsi_root "$netroot" || return
