From 71909cc72ccd11af9c542b134b112b8d6f8c5211 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 4 Jun 2012 11:33:29 +0200
Subject: [PATCH] dracut.sh: do not rely on the presence of lib64, check with
 ldd

---
 dracut.sh |   27 +++++++++++++++++----------
 1 file changed, 17 insertions(+), 10 deletions(-)

diff --git a/dracut.sh b/dracut.sh
index 1048e75..44dcc94 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -512,16 +512,23 @@ ddebug "Executing $0 $dracut_args"
 }
 
 # Detect lib paths
-[[ $libdir ]] || for libdir in /lib64 /lib; do
-    [[ -d $libdir ]] && libdirs+=" $libdir" && break
-done || {
-    dfatal 'No lib directory?!!!'
-    exit 1
-}
-
-[[ $usrlibdir ]] || for usrlibdir in /usr/lib64 /usr/lib; do
-    [[ -d $usrlibdir ]] && libdirs+=" $usrlibdir" && break
-done || dwarn 'No usr/lib directory!'
+if ! [[ $libdir ]] || ! [[ $usrlibdir ]] ; then
+    if strstr "$(ldd /bin/sh)" "/lib64/" &>/dev/null \
+        && [[ -d /lib64 ]]; then
+        libdir=/lib64
+        usrlibdir=/usr/lib64
+    else
+        libdir=/lib
+        usrlibdir=/usr/lib
+    fi
+    for i in $libdir $usrlibdir; do
+        if [[ -d $i ]]; then
+            libdirs+=" $i"
+        else
+            dwarn 'No $i directory??!!'
+        fi
+    done
+fi
 
 # This is kinda legacy -- eventually it should go away.
 case $dracutmodules in
