From 7f1d77d9fb27f678f50aa6ad1baac7d28e8e4f5f Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 26 Sep 2012 11:58:51 +0200
Subject: [PATCH] TEST-16-DMSQUASH: reenable test

---
 test/TEST-16-DMSQUASH/test.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/TEST-16-DMSQUASH/test.sh b/test/TEST-16-DMSQUASH/test.sh
index 4dc133c..9c4f356 100755
--- a/test/TEST-16-DMSQUASH/test.sh
+++ b/test/TEST-16-DMSQUASH/test.sh
@@ -11,7 +11,7 @@ test_check() {
         echo "python-imgcreate not installed"
 	return 1
     fi
-    return 1
+    return 0
 }
 
 test_run() {
