From eb74dc4b0752731e5bf6929bec06d16aebf93b03 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 12 Jun 2012 11:06:58 -0400
Subject: [PATCH] dracut-functions: fixed return status of instmods()

---
 dracut-functions.sh |   24 ++++++++++++++++--------
 dracut.sh           |    4 ++++
 2 files changed, 20 insertions(+), 8 deletions(-)

diff --git a/dracut-functions.sh b/dracut-functions.sh
index ccf3ba6..7b1c8dc 100755
--- a/dracut-functions.sh
+++ b/dracut-functions.sh
@@ -1054,17 +1054,17 @@ install_kmod_with_fw() {
 # rest of args = arguments to modprobe
 # _fderr specifies FD passed from surrounding scope
 for_each_kmod_dep() {
-    local _func=$1 _kmod=$2 _cmd _modpath _options _found=0
+    local _func=$1 _kmod=$2 _cmd _modpath _options _ret
     shift 2
     modprobe "$@" --ignore-install --show-depends $_kmod 2>&${_fderr} | (
         while read _cmd _modpath _options; do
             [[ $_cmd = insmod ]] || continue
             $_func ${_modpath} || exit $?
-            _found=1
         done
-        [[ $_found -eq 0 ]] && exit 1
         exit 0
     )
+    _ret=$?
+    return $_ret
 }
 
 # filter kernel modules to install certain modules that meet specific
@@ -1130,16 +1130,19 @@ instmods() {
 
     function inst1mod() {
         local _mod="$1"
+        local _ret=0
         case $_mod in
             =*)
                 if [ -f $srcmods/modules.${_mod#=} ]; then
                     ( [[ "$_mpargs" ]] && echo $_mpargs
                       cat "${srcmods}/modules.${_mod#=}" ) \
                     | instmods
+                    ((_ret+=$?))
                 else
                     ( [[ "$_mpargs" ]] && echo $_mpargs
                       find "$srcmods" -path "*/${_mod#=}/*" -printf '%f\n' ) \
                     | instmods
+                    ((_ret+=$?))
                 fi
                 ;;
             --*) _mpargs+=" $_mod" ;;
@@ -1152,13 +1155,13 @@ instmods() {
 
                 if [[ $omit_drivers ]] && [[ "$1" =~ $omit_drivers ]]; then
                     dinfo "Omitting driver ${_mod##$srcmods}"
-                    return
+                    return 0
                 fi
                 # If we are building a host-specific initramfs and this
                 # module is not already loaded, move on to the next one.
                 [[ $hostonly ]] && ! grep -qe "\<${_mod//-/_}\>" /proc/modules \
                     && ! echo $add_drivers | grep -qe "\<${_mod}\>" \
-                    && return
+                    && return 0
 
                 # We use '-d' option in modprobe only if modules prefix path
                 # differs from default '/'.  This allows us to use Dracut with
@@ -1173,6 +1176,7 @@ instmods() {
                 ((_ret+=$?))
                 ;;
         esac
+        return $_ret
     }
 
     function instmods_1() {
@@ -1180,19 +1184,23 @@ instmods() {
         if (($# == 0)); then  # filenames from stdin
             while read _mod; do
                 inst1mod "${_mod%.ko*}"
+                ((_ret+=$?))
             done
         fi
         while (($# > 0)); do  # filenames as arguments
             inst1mod ${1%.ko*}
+            ((_ret+=$?))
             shift
         done
         return $_ret
     }
 
     local _filter_not_found='FATAL: Module .* not found.'
+    local _ret=0;
     # Capture all stderr from modprobe to _fderr. We could use {var}>...
     # redirections, but that would make dracut require bash4 at least.
-    eval "( instmods_1 \"\$@\" ) ${_fderr}>&1" \
-    | while read line; do [[ "$line" =~ $_filter_not_found ]] || echo $line;done | derror
-    return $?
+    eval "( instmods_1 \"\$@\" ) ${_fderr}>&1" | \
+        while read line; do [[ "$line" =~ $_filter_not_found ]] || echo $line;done  | derror
+    _ret=$?
+    return $_ret
 }
diff --git a/dracut.sh b/dracut.sh
index 44dcc94..eb58f78 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -26,6 +26,9 @@
 # store for logging
 dracut_args="$@"
 
+# we want the whole pipe to succeed
+set -o pipefail
+
 usage() {
 #                                                       80x25 linebreak here ^
     cat << EOF
@@ -729,6 +732,7 @@ done
 unset moddir
 
 for i in $modules_loaded; do
+    mkdir -p $initdir/lib/dracut
     echo "$i" >> $initdir/lib/dracut/modules.txt
 done
 
